### PR TITLE
Remove Control.Carrier and move its contents to Control.Carrier.Class.

### PR DESCRIPTION
**Pull request by patrickt**
This module was designed when we were still endeavoring to avoid .hs-boot files. We have since abandoned this particular pursuit, since .hs-boot files are necessary to provide the kind of carrier instances we want. This means we can nuke Control.Carrier and provide its useful exports through `Control.Carrier.Class`, which we may rename in the future to `Control.Algebra`.

### CODE REVIEW

File: src/Control/Carrier/Pure.hs-boot [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module Control.Carrier.Pure
```
Comment (patrickt) [inline src/Control/Carrier/Pure.hs-boot]: I don’t love that we had to add this, but it keeps the change smaller than the other options, and preserves the useful property that an exported `run` from a given carrier means that you can do useful work with a single import.

File: src/Control/Carrier/Class.hs [hunk 2]
```diff
@@ -1,8 +1,15 @@
-{-# LANGUAGE DeriveFunctor, EmptyCase, FlexibleInstances, FunctionalDependencies, TypeOperators, UndecidableInstances #-}
+{-# LANGUAGE ConstraintKinds, DeriveFunctor, EmptyCase, FlexibleInstances, FunctionalDependencies, TypeOperators, UndecidableInstances #-}
 module Control.Carrier.Class
 ( Carrier(..)
+, Has
+, send
+  -- * Re-exports
+, (:+:) (..)
+, run
+, module Control.Effect.Class
```
Comment (robrix) [inline src/Control/Carrier/Class.hs]: NB: we had also re-exported `PureC` from `Control.Carrier`. I don’t _think_ that’s super important for modules defining either effects or carriers, but it is a difference. (Effects merely consuming effects or carriers wouldn’t have been importing this anyway.)

File: src/Control/Carrier/Class.hs [hunk 3]
```diff
@@ -36,6 +43,19 @@ class (HFunctor sig, Monad m) => Carrier sig m | m -> sig where
...
+--
+-- 1. Due to [a problem with recursive type families](https://gitlab.haskell.org/ghc/ghc/issues/8095), this can lead to significantly slower compiles.
+--
+-- 2. It defeats @ghc@’s warnings for redundant constraints, and thus can lead to a proliferation of redundant constraints as code is changed.
+type Has eff sig m = (Members eff sig, Carrier sig m)
+
+-- | Construct a request for an effect to be interpreted by some handler later on.
+send :: (Member eff sig, Carrier sig m) => eff m a -> m a
+send = eff . inj
+{-# INLINE send #-}
```
Comment (robrix) [inline src/Control/Carrier/Class.hs]: Could you please add another line of whitespace here to preserve the two-line gap before the next section? 🙏

File: src/Control/Carrier/Pure.hs-boot (hunk 1 above)
Comment (robrix) [reply src/Control/Carrier/Pure.hs-boot]: Yeah, I think this is fine 👍

**Pull request merged**